homeassistant/components/habitica/config_flow.py

```diff
@@ -0,0 +1,84 @@
+"""Config flow for habitica integration."""
+import logging
+
+from aiohttp import ClientResponseError
+from habitipy.aio import HabitipyAsync
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_API_KEY, CONF_NAME, CONF_URL
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_API_USER, DEFAULT_URL, DOMAIN  # pylint: disable=unused-import
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_USER): str,
+        vol.Required(CONF_API_KEY): str,
+        vol.Optional(CONF_NAME): str,
+        vol.Optional(CONF_URL, default=DEFAULT_URL): str,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    async def authenticate(username, password, url=DEFAULT_URL) -> bool:
```

Comment:
We don't need the `authenticate` function. We can inline that with the `validate_input` function.